web/modules/custom/joinup_user/src/Form/UnsubscribeFromAllCollectionsForm.php

```diff
@@ -0,0 +1,281 @@
+      $container->get('og.membership_manager'),
+      $container->get('renderer'),
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'unsubscribe_from_all_collections_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Unsubscribe from all collections');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDescription() {
+    return $this->t('Are you sure you want to unsubscribe from all collections?<br />You will stop receiving news and updates from all collections, including those you are a facilitator in.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    return Url::fromRoute('joinup_subscription.subscriptions_page');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    if ($memberships = $this->getUserMembershipIds()) {
+      $memberships = $this->entityTypeManager->getStorage('og_membership')->loadMultiple($memberships);
+      $labels = array_map(function (OgMembershipInterface $membership) {
+        return $membership->getGroup()->label();
+      }, $memberships);
+      asort($labels);
+      $form = parent::buildForm($form, $form_state);
+      $form['information'] = [
+        '#type' => 'item',
+        '#markup' => t('You are currently registered to be notified for the following collections:'),
+        '#tree' => TRUE,
+        'items' => [
+          '#theme' => 'item_list',
+          '#items' => $labels,
+        ],
+      ];
+    }
+    else {
+      $form['help'] = [
+        '#type' => 'item',
+        '#markup' => t('You do not have any collections to unsubscribe from.'),
+      ];
+
+      $form['return'] = [
+        '#type' => 'submit',
+        '#value' => $this->t('Return'),
+        '#submit' => ['::returnSubmit'],
+      ];
+    }
+    return $form;
+  }
+
+  /**
+   * Submit callback for the return button.
+   *
+   * Returns the user to the same destination as the cancel URL.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   */
+  public function returnSubmit(array &$form, FormStateInterface $form_state): void {
+    $return_url = $this->getCancelUrl();
+    $form_state->setRedirect($return_url->getRouteName(), $return_url->getRouteParameters());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $membership_ids = $this->getUserMembershipIds();
+    $redirect_url = $this->getCancelUrl();
+    $form_state->setRedirect($redirect_url->getRouteName(), $redirect_url->getRouteParameters());
+
+    $operations = [];
+    foreach ($membership_ids as $membership_id) {
+      $operations[] = [
+        '\Drupal\joinup_user\Form\UnsubscribeFromAllCollectionsForm::membershipUnsubscribe',
```

Comment:
As of PHP 5.5.0 we have a constant that allows us to get the fully qualified name of the class w/o needing to hardocode it: `UnsubscribeFromAllCollectionsForm::class . '::membershipUnsubscribe'`